app/controllers/comments_controller.rb

```diff
@@ -1,22 +1,40 @@
 # frozen_string_literal: true
 
 class CommentsController < ApplicationController
-  before_filter :set_comment, only: [:show, :update, :destroy]
-  before_filter :set_article, only: [:index, :create]
+  before_action :set_comment, only: [:show, :update, :destroy]
+  before_action :set_article, only: [:index, :create]
 
   def index
+    @comments = @article.comments
+    render json: { comments: @comments }
   end
 
   def show
+    render json: @comment
   end
 
   def create
+    @comment = Comment.new(comment_params)
+    @comment.article = @article
```

Comment:
Un-nesting routes would move this line (article) into the previous line as part of the request body.

In other words, instead of sending the article ID as part of the URL, the article ID is sent by the client as part of "comment_params"